src/backend/cdb/cdbmutate.c

```diff
@@ -75,8 +75,18 @@ typedef struct ApplyMotionState
 								 * plan_tree_walker/mutator */
 	int			nextMotionID;
 	int			sliceDepth;
-	bool		containMotionNodes;
 	HTAB	   *planid_subplans; /* hash table for InitPlanItem */
+
+	/* Context for ModifyTable to elide Explicit Redistribute Motion */
+	bool		mtIsChecking;		/* True if we encountered ModifyTable
+									 * node with UPDATE/DELETE and we plan
+									 * to insert Explicit Motions. */
+	Bitmapset  *mtResultRtis; 		/* Indexes into rtable for relations to
+									 * be modified. Only valid if mtIsChecking
+									 * is true. */
+	bool		needExplicitMotion;
```

Comment:
Is this field redundant?

Reply:
https://github.com/arenadata/gpdb/pull/666#discussion_r1503597642